mmv1/third_party/terraform/resources/resource_compute_router_interface.go.erb

```diff
@@ -47,28 +47,40 @@ func resourceComputeRouterInterface() *schema.Resource {
 			},
 			"vpn_tunnel": {
 				Type:             schema.TypeString,
-				ConflictsWith:    []string{"interconnect_attachment"},
 				Optional:         true,
 				ForceNew:         true,
 				DiffSuppressFunc: compareSelfLinkOrResourceName,
-				AtLeastOneOf:     []string{"vpn_tunnel", "interconnect_attachment", "ip_range"},
```

Comment:
I suspect we want to keep the `AtLeastOneOf` but add `subnetwork` to it as well, rather than remove it! This ensures that users have specified a meaningful resource w/o a round trip to the API.

Reply:
I agree we should bring it back. Anyway, the [gcloud reference](https://cloud.google.com/sdk/gcloud/reference/compute/routers/add-interface) says "Exactly one of these must be specified:" between `vpn_tunnel`, `interconnect_attachment`, `subnetwork`. Instead, `ip-address` and `mask-length` (which are ip_range here) should be optional. I've noticed that in some tests this throws an error, given we're just specifying `ip_range` and none of the other three. I'll do the mod and let the tests fail. Once we know what tests are failing, let's try to work on a proposal to modify them.